.circleci/config.yml

```diff
@@ -131,12 +125,9 @@ jobs:
             - run:
                 name: Build Haskell
                 command: source ~/.profile && make kademlia
-            - run:
-                name: Enable snarks --  make withsnark
-                command: make withsnark
             - run:
                 name: Build OCaml
-                command: eval `opam config env` && make build 2>&1 | tee /tmp/buildocaml.log
+                command: eval `opam config env` && DUNE_PROFILE=test_snark make build 2>&1 | tee /tmp/buildocaml.log
```

Comment:
I'm 99% certain we need `DUNE_PROFILE` to be passed to every `make` invocation below or it won't work.

Reply:
IIUC, the only reason we're not seeing `dune` rebuilding for the subsequent `dune exec`s that the automation does is that dune doesn't keep track of the profile and invalidate old artifacts ... is that correct @nholland94?

Reply:
@cmr AFAIK, dune build will properly invalidate builds with profile changes but dune exec will not. I'm not really sure why, but this is what I've noticed locally. If you always dune build with the correct profile before running dune exec, it should work.